src/main/java/net/minecraftforge/client/RecipeBookManager.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.client;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import net.minecraft.client.RecipeBookCategories;
+import net.minecraft.world.inventory.RecipeBookType;
+import net.minecraft.world.item.crafting.Recipe;
+import net.minecraft.world.item.crafting.RecipeType;
+import net.minecraftforge.client.event.RegisterRecipeBookCategoriesEvent;
+import net.minecraftforge.fml.ModLoader;
+import net.minecraftforge.fml.ModLoadingContext;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+import static net.minecraft.client.RecipeBookCategories.*;
+
+/**
+ * Manager for {@link RecipeBookType recipe book types} and {@link RecipeBookCategories categories}.
+ * <p>
+ * Provides a recipe category lookup.
+ */
+public class RecipeBookManager
+{
+    // Not using ConcurrentHashMap here because it's slower for lookups, so we only use it during init
+    private static final Map<RecipeBookCategories, List<RecipeBookCategories>> AGGREGATE_CATEGORIES = new HashMap<>();
+    private static final Map<RecipeBookType, List<RecipeBookCategories>> TYPE_CATEGORIES = new HashMap<>();
+    private static final Map<RecipeType<?>, Function<Recipe<?>, RecipeBookCategories>> RECIPE_CATEGORY_LOOKUPS = new HashMap<>();
+
+    //Unmodifiable view, the nested lists are immutable
```

Comment:
Formatting: space between comment header and comment body.
```suggestion
    // Unmodifiable view, the nested lists are immutable
```